### PR TITLE
Added iOS specific section for OIDC Auth provider

### PR DESCRIPTION
**Pull request by wooj2**
Added iOS specific section for OIDC Auth provider
Screen Shot 2020-10-29 at 7 36 09 PM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/lib/datastore/fragments/ios/setup-auth-rules/owner_based_auth_oidc.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+When using a 3rd party OIDC auth provider, you are required to provide an instance of `APIAuthProviderFactory` which is capable of creating an instance of a class that conforms to `AmplifyOIDCAuthProvider`.  The responsibility of the `AmplifyOIDCAuthProvider` is to return the JWT token that was provided by your OIDC provider.  To do this:
+* Create a subclass of `APIAuthProviderFactory`
```
Comment (jamesonwilliams) [inline docs/lib/datastore/fragments/ios/setup-auth-rules/owner_based_auth_oidc.md:2]:
```suggestion
When using a third-party OIDC auth provider, you are required to provide an instance of `APIAuthProviderFactory`. Your factory implementation must be capable of creating objects conforming to the `AmplifyOIDCAuthProvider` protocol.  The responsibility of the `AmplifyOIDCAuthProvider` is to return the JWT token that was provided by your OIDC provider. To do this:
* Create a subclass of `APIAuthProviderFactory`
```

File: docs/lib/datastore/fragments/ios/setup-auth-rules/owner_based_auth_oidc.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+class MyAPIAuthProviderFactory: APIAuthProviderFactory {
+    let myAuthProvider = MyOIDCAuthProvider()
+
+    override func oidcAuthProvider() -> AmplifyOIDCAuthProvider? {
+        return myAuthProvider
+    }
+}
+```
+
+* Implement your class which conforms to `AmplifyOIDCAuthProvider`
```
Comment (jamesonwilliams) [inline docs/lib/datastore/fragments/ios/setup-auth-rules/owner_based_auth_oidc.md:13]:
```suggestion
* Implement your class which conforms to `AmplifyOIDCAuthProvider`:
```

File: docs/lib/datastore/fragments/ios/setup-auth-rules/owner_based_auth_oidc.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+```swift
+class MyOIDCAuthProvider : AmplifyOIDCAuthProvider {
+    func getLatestAuthToken() -> Result<String, Error> {
+       ....
+    }
+}
+```
+* Finally, register your instance of `APIAuthProviderFactory` prior to calling `Amplify.configure()`
```
Comment (jamesonwilliams) [inline docs/lib/datastore/fragments/ios/setup-auth-rules/owner_based_auth_oidc.md:21]:
```suggestion
* Finally, register your instance of `APIAuthProviderFactory` prior to calling `Amplify.configure()`:
```

**Review by jamesonwilliams — APPROVED** [review]

**Review by richardmcclellan — COMMENTED** [review]
I'd suggest swapping the AuthProvider step with the ProviderFactory step, so that AuthProvider is first, since the factory depends on the provider.  Otherwise, looks good, and I'll add similar docs for Android.

File: docs/lib/datastore/fragments/ios/setup-auth-rules/owner_based_auth_oidc.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+When using a third-party OIDC auth provider, you are required to provide an instance of `APIAuthProviderFactory`. Your factory implementation must be capable of creating objects conforming to the `AmplifyOIDCAuthProvider` protocol.  The responsibility of the `AmplifyOIDCAuthProvider` is to return the JWT token that was provided by your OIDC provider. To do this:
```
Comment (renebrandel) [inline docs/lib/datastore/fragments/ios/setup-auth-rules/owner_based_auth_oidc.md:1]: Remove double whitespaces

**Review by renebrandel — APPROVED** [review]
nitpick on double whitespaces but otherwise looks fine.

**Pull request merged**